port/unix/omrsock.c

```diff
@@ -76,8 +76,35 @@ omrsock_getaddrinfo_create_hints(struct OMRPortLibrary *portLibrary, omrsock_add
 
 int32_t
 omrsock_getaddrinfo(struct OMRPortLibrary *portLibrary, char *node, char *service, omrsock_addrinfo_t hints, omrsock_addrinfo_t result)
-{
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+{	
+	omr_os_addrinfo *addrInfoResults = NULL;
+	omr_os_addrinfo *addrInfoHints = NULL;
+	int32_t count = 0;
+	
+	if (NULL == result) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	
+	memset(result, 0, sizeof(struct OMRAddrInfoNode));
+
+	if (NULL != hints) {
+		addrInfoHints = (omr_os_addrinfo *)hints->addrInfo;
+	}
+
+	if (0 != getaddrinfo(node, service, addrInfoHints, &addrInfoResults)) {
+		return OMRPORT_ERROR_SOCK_ADDRINFO_FAILED;
+	}
+
+	result->addrInfo = addrInfoResults;
+
+	while (NULL != addrInfoResults->ai_next) {
+		count++;
+		addrInfoResults = addrInfoResults->ai_next;
+	}
+	/* Account the first entry */
+	result->length = count++;
```

Comment:
I would initialize count to 1 before the while loop, rather than adding 1 at the end, it makes more sense to me.